### PR TITLE
Add check for undefined cert

### PR DESCRIPTION
**Pull request by Tsuesun**
Query will return `undefined`  if a certificate is not present, this will cause a type error when calling` .replace()` on it in the `.map`

This slightly modifies the logic so that a more meaningful error is thrown.

### CODE REVIEW

**Comment by compwright** [comment]
@Tsuesun thanks for this contribution. I see there are now a number of conflicts, please rebase against the latest master, or if preferred resubmit as a new PR.

**Review by compwright — CHANGES_REQUESTED** [review]
Thanks for this contribution! I see there are now a number of conflicts, please rebase against the latest master, or if preferred resubmit as a new PR

**Comment by Tsuesun** [comment]
> Thanks for this contribution! I see there are now a number of conflicts, please rebase against the latest master, or if preferred resubmit as a new PR

Hi @compwright, all done. Checked the tests and linter and both passed.

File: src/MetadataReader.js [hunk 1]
```diff
@@ -131,7 +135,7 @@ export class MetadataReader {
 
   get encryptionCert () {
     try {
-      return this.encryptionCerts[0].replace(/[\r\n\t\s]/gm, '')
```
Comment (compwright) [inline src/MetadataReader.js:134]: Why this change?

File: src/MetadataReader.js [hunk 2]
```diff
@@ -156,7 +164,7 @@ export class MetadataReader {
 
   get signingCert () {
     try {
-      return this.signingCerts[0].replace(/[\r\n\t\s]/gm, '')
```
Comment (compwright) [inline src/MetadataReader.js:159]: Why this change?

File: src/MetadataReader.js (hunk 1 above)
Comment (compwright) [inline src/MetadataReader.js:134]: Why this change?

**Review by compwright — COMMENTED** [review]
Thanks so much for bringing this PR up to date! A couple questions on two changes that aren't clear to me why they are needed. Can you explain?

File: src/MetadataReader.js (hunk 2 above)
Comment (Tsuesun) [reply src/MetadataReader.js:159]: `get signingCerts()` and `get encryptionCerts()` return an array that has already been normalised with `certs.map((node) => node.firstChild.data.replace(/[\r\n\t\s]/gm, ''))` so this replace seems to be redundant and not serve a purpose. I confirmed that by making sure these tests pass: https://github.com/compwright/passport-saml-metadata/blob/master/src/MetadataReader.test.js#L69

**Review by compwright — APPROVED** [review]

**Comment by compwright** [comment]
@Tsuesun released, v3.1.1
